config/main.py

```diff
@@ -1068,6 +1174,75 @@ def speed(ctx, interface_name, interface_speed, verbose):
         command += " -vv"
     run_command(command, display_cmd=verbose)
 
+
+#
+# 'breakout' subcommand
+#
+
+@interface.command()
+@click.argument('interface_name', metavar='<interface_name>', required=True)
+@click.argument('mode', required=True, type=click.STRING, autocompletion=_get_option)
+@click.option('-y', '--yes', is_flag=True, callback=abort_if_false, expose_value=False, prompt='Do you want to Breakout the port, continue?')
+@click.option('-v', '--verbose', is_flag=True, help="Enable verbose output")
+def breakout(interface_name, mode, verbose):
+
+    """ Set interface breakout mode """
+    breakout_file_input = readJsonFile(BREAKOUT_CFG_FILE)
+    target_breakout_mode = mode
+    if interface_name in breakout_file_input:
+
+        # Check whether target breakout mode is available for the user-selected interface or  not
+        if target_breakout_mode not in breakout_file_input[interface_name]["breakout_modes"]:
+            click.secho('[ERROR] Target mode {} is not available for the port {}'. format(target_breakout_mode, interface_name), fg='red')
+            sys.exit(1)
+
+        """ Interface Deletion Logic """
+        config_db = ConfigDBConnector()
+        config_db.connect()
+        port_dict = config_db.get_table('PORT')
+        current_breakout_dict = config_db.get_table('BREAKOUT_CFG')
+        if not port_dict:
+            click.echo("port_dict is None!")
+            raise click.Abort()
+        if interface_name in port_dict.keys():
+                running_breakout_mode = current_breakout_dict[interface_name]["current_brkout_mode"]
+                click.echo()
+                click.echo("\nRunning Breakout Mode : {} \nTarget Breakout Mode : {}".format(running_breakout_mode, target_breakout_mode))
+                intended_delete_interface_speed_dict = _get_child_interface_speed_dict(interface_name, running_breakout_mode)
+                if intended_delete_interface_speed_dict:
+                        click.echo("\nPorts to be deleted : \n {}".format(json.dumps(intended_delete_interface_speed_dict, indent=4)))
+                else:
+                     click.echo("port_dict is None!")
+                     raise click.Abort()
+
+        """ Interface Addition Logic """
+        intended_add_interface_speed_dict = _get_child_interface_speed_dict(interface_name, target_breakout_mode)
+        if intended_add_interface_speed_dict:
+                click.echo("Ports to be added : \n {}".format(json.dumps(intended_add_interface_speed_dict, indent=4)))
+        else:
+                click.echo("port_dict is None!")
+                raise click.Abort()
+
+        """ Special Case: Dont delete those ports  where the current mode and speed of the parent port remains unchanged
+                          to limit the traffic impact
+        """
+        click.secho("\nAfter running Logic to limit the impact", fg="cyan", underline=True)
```

Comment:
Let's make this optimization as an option too.

Reply:
will change the code once backend create/delete API is available.